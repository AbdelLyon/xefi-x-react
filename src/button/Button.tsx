import type {
  ButtonProps as HeroUIButtonProps,
  PressEvent,
} from "@heroui/react"
import { Button as HeroUIButton, Spinner } from "@heroui/react"
import type { JSX, ReactNode } from "react"
import { forwardRef } from "react"
import { mergeTailwindClasses } from "@/utils"
import type {
  Color,
  Variant,
  Size,
  BaseComponentProps,
  AccessibilityProps,
} from "@/types"

/**
 * Enhanced Button props extending HeroUI Button with additional features
 */
export interface ButtonProps
  extends Omit<
      HeroUIButtonProps,
      | "color"
      | "variant"
      | "size"
      | "aria-label"
      | "aria-disabled"
      | "onPress"
      | "onClick"
      | "startContent"
      | "endContent"
    >,
    BaseComponentProps,
    Pick<
      AccessibilityProps,
      "aria-label" | "aria-labelledby" | "aria-describedby"
    > {
  /** Visual variant of the button */
  variant?: Variant
  /** Color theme of the button */
  color?: Color
  /** Size of the button */
  size?: Size
  /** Whether button is in loading state */
  loading?: boolean
  /** Icon to display on the left side */
  leftIcon?: ReactNode
  /** Icon to display on the right side */
  rightIcon?: ReactNode
  /** Whether button should take full width */
  fullWidth?: boolean
  /** Click handler */
  onClick?: (event: PressEvent) => void
  /** Custom loading spinner */
  loadingSpinner?: ReactNode
  /** Loading text to show alongside spinner */
  loadingText?: string
  /** Whether to show ripple effect */
  disableRipple?: boolean
  /** Animation type for the button */
  disableAnimation?: boolean
}

/**
 * Enhanced Button component built on top of HeroUI Button
 *
 * @example
 * ```tsx
 * // Basic button
 * <Button color="primary">Click me</Button>
 *
 * // Button with icons
 * <Button
 *   leftIcon={<PlusIcon />}
 *   rightIcon={<ChevronIcon />}
 *   variant="bordered"
 * >
 *   Add Item
 * </Button>
 *
 * // Loading button
 * <Button loading loadingText="Processing...">
 *   Submit
 * </Button>
 * ```
 */
export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      variant = "solid",
      color = "primary",
      size = "md",
      loading = false,
      leftIcon,
      rightIcon,
      fullWidth = false,
      children,
      disabled,
      className,
      onClick,
      loadingSpinner,
      loadingText,
      disableRipple = false,
      disableAnimation = false,
      ...props
    },
    ref
  ): JSX.Element => {
    const isDisabled = disabled || loading

    const handlePress = (e: PressEvent): void => {
      if (loading) {
        return
      }
      e.continuePropagation()
      onClick?.(e)
    }

    // Custom loading spinner or default
    const spinner = loadingSpinner || (
      <Spinner
        size={size === "sm" ? "sm" : "md"}
        color="current"
        classNames={{
          circle1: "border-b-current",
          circle2: "border-b-current",
        }}
      />
    )

    // Determine content based on loading state
    const buttonContent = loading && loadingText ? loadingText : children

    return (
      <HeroUIButton
        ref={ref}
        variant={variant}
        onPress={handlePress}
        color={color}
        size={size as "sm" | "md" | "lg"}
        isDisabled={isDisabled}
        disableRipple={disableRipple || loading}
        disableAnimation={disableAnimation}
        startContent={loading ? spinner : leftIcon}
        endContent={!loading ? rightIcon : undefined}
        className={mergeTailwindClasses(
          fullWidth && "w-full",
          loading && "cursor-wait relative",
          className
        )}
        {...props}
      >
        {buttonContent}
      </HeroUIButton>
    )
  }
)

Button.displayName = "Button"
