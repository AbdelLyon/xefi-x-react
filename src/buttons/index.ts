import "@/index.css";
export { Buttons, type ButtonsProps } from "@/buttons/Buttons";
