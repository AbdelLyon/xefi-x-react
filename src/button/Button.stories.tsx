import type { Meta, StoryObj } from "@storybook/react-vite"
import {
  IconPlus,
  IconDownload,
  IconHeart,
  IconStar,
  IconShare,
  IconSettings,
  IconBell,
  IconUser,
  IconShoppingCart,
  IconSearch,
  IconArrowRight,
  IconCheck,
} from "@tabler/icons-react"
import { Button } from "./Button"

const meta: Meta<typeof Button> = {
  title: "Components/Button",
  component: Button,
  parameters: {
    layout: "centered",
    docs: {
      description: {
        component:
          "**Button** - Composant bouton élégant avec support complet dark/light mode, animations fluides et personnalisations avancées.",
      },
    },
  },
  argTypes: {
    variant: {
      control: "select",
      options: [
        "solid",
        "bordered",
        "light",
        "flat",
        "faded",
        "shadow",
        "ghost",
      ],
      description: "Style visuel du bouton",
    },
    color: {
      control: "select",
      options: [
        "primary",
        "secondary",
        "success",
        "warning",
        "danger",
        "default",
      ],
      description: "Couleur thématique",
    },
    size: {
      control: "select",
      options: ["sm", "md", "lg"],
      description: "Taille du bouton",
    },
    radius: {
      control: "select",
      options: ["none", "sm", "md", "lg", "full"],
      description: "Radius des coins (défaut: md)",
    },
    loading: {
      control: "boolean",
      description: "État de chargement",
    },
    disabled: {
      control: "boolean",
      description: "Bouton désactivé",
    },
    fullWidth: {
      control: "boolean",
      description: "Pleine largeur",
    },
    children: {
      control: "text",
      description: "Contenu du bouton",
    },
  },
  tags: ["autodocs"],
}

export default meta
type Story = StoryObj<typeof meta>

// Story par défaut - utilise le radius par défaut (md)
export const Default: Story = {
  args: {
    children: "Bouton par défaut",
    color: "primary",
  },
}

// Test spécifique du radius par défaut
export const RadiusComparison: Story = {
  render: () => (
    <div className="max-w-4xl space-y-8 p-6">
      <div>
        <h3 className="mb-4 text-lg font-semibold text-foreground">
          Test du radius par défaut
        </h3>
        <div className="flex flex-wrap items-center gap-4">
          <Button color="primary">Sans radius explicite (défaut: md)</Button>
          <Button color="primary" radius="md">
            Avec radius="md" explicite
          </Button>
        </div>
        <p className="mt-2 text-sm text-default-500">
          ↑ Ces deux boutons devraient avoir exactement la même apparence
        </p>
      </div>

      <div>
        <h3 className="mb-4 text-lg font-semibold text-foreground">
          Tous les radius disponibles avec style inline forcé
        </h3>
        <div className="flex flex-wrap items-center gap-4">
          <Button color="primary" radius="none">
            none (0px)
          </Button>
          <Button color="primary" radius="sm">
            sm (4px)
          </Button>
          <Button color="primary" radius="md">
            md (6px) - défaut
          </Button>
          <Button color="primary" radius="lg">
            lg (8px)
          </Button>
          <Button color="primary" radius="full">
            full (9999px)
          </Button>
        </div>
        <div className="mt-4 rounded-lg bg-default-100 p-4">
          <p className="text-sm text-default-600">
            <strong>Debug :</strong> Chaque bouton a maintenant un style inline
            qui force le border-radius. Ouvrez les outils de développement pour
            voir <code>style="border-radius: Xpx"</code> sur chaque bouton.
          </p>
        </div>
      </div>

      <div>
        <h3 className="mb-4 text-lg font-semibold text-foreground">
          Test avec différentes variantes
        </h3>
        <div className="grid grid-cols-5 gap-4">
          {(["none", "sm", "md", "lg", "full"] as const).map((radiusValue) => (
            <div key={radiusValue} className="space-y-2">
              <h4 className="text-center text-sm font-medium">{radiusValue}</h4>
              <Button
                color="primary"
                radius={radiusValue}
                size="sm"
                className="w-full"
              >
                Solid
              </Button>
              <Button
                color="primary"
                radius={radiusValue}
                variant="bordered"
                size="sm"
                className="w-full"
              >
                Bordered
              </Button>
              <Button
                color="primary"
                radius={radiusValue}
                variant="light"
                size="sm"
                className="w-full"
              >
                Light
              </Button>
            </div>
          ))}
        </div>
      </div>
    </div>
  ),
  parameters: {
    layout: "fullscreen",
  },
}

// Showcase moderne
export const ModernShowcase: Story = {
  render: () => (
    <div className="mx-auto max-w-6xl p-8">
      <div className="mb-8">
        <h2 className="mb-2 text-3xl font-bold text-foreground">
          Collection de Boutons Modernes
        </h2>
        <p className="text-default-500">
          Boutons élégants avec design contemporain et radius par défaut
        </p>
      </div>

      <div className="grid grid-cols-1 gap-8 md:grid-cols-2 lg:grid-cols-4">
        {/* Actions Principales */}
        <div className="space-y-4">
          <h4 className="text-sm font-semibold uppercase tracking-wider text-foreground">
            Actions Principales
          </h4>
          <Button
            color="primary"
            variant="shadow"
            size="lg"
            leftIcon={<IconPlus size={18} />}
            className="w-full font-medium"
          >
            Créer
          </Button>
          <Button
            color="success"
            variant="solid"
            leftIcon={<IconDownload size={16} />}
            className="w-full"
          >
            Télécharger
          </Button>
          <Button
            color="secondary"
            variant="flat"
            leftIcon={<IconShare size={16} />}
            className="w-full"
          >
            Partager
          </Button>
        </div>

        {/* Interactifs */}
        <div className="space-y-4">
          <h4 className="text-sm font-semibold uppercase tracking-wider text-foreground">
            Interactifs
          </h4>
          <Button
            color="danger"
            variant="bordered"
            leftIcon={<IconHeart size={16} />}
            className="w-full"
          >
            Favoris
          </Button>
          <Button
            color="warning"
            variant="light"
            leftIcon={<IconStar size={16} />}
            className="w-full"
          >
            Évaluer
          </Button>
          <Button
            color="primary"
            variant="ghost"
            leftIcon={<IconBell size={16} />}
            className="w-full"
          >
            Notifications
          </Button>
        </div>

        {/* Utilitaires */}
        <div className="space-y-4">
          <h4 className="text-sm font-semibold uppercase tracking-wider text-foreground">
            Utilitaires
          </h4>
          <Button
            color="default"
            variant="bordered"
            leftIcon={<IconSettings size={16} />}
            className="w-full"
          >
            Paramètres
          </Button>
          <Button
            color="primary"
            variant="flat"
            leftIcon={<IconUser size={16} />}
            className="w-full"
          >
            Profil
          </Button>
          <Button
            color="success"
            variant="light"
            leftIcon={<IconCheck size={16} />}
            className="w-full"
          >
            Valider
          </Button>
        </div>

        {/* E-commerce */}
        <div className="space-y-4">
          <h4 className="text-sm font-semibold uppercase tracking-wider text-foreground">
            E-commerce
          </h4>
          <Button
            color="primary"
            variant="shadow"
            leftIcon={<IconShoppingCart size={16} />}
            className="w-full font-medium"
          >
            Ajouter au panier
          </Button>
          <Button
            color="default"
            variant="bordered"
            leftIcon={<IconSearch size={16} />}
            className="w-full"
          >
            Rechercher
          </Button>
          <Button
            color="secondary"
            variant="solid"
            rightIcon={<IconArrowRight size={16} />}
            className="w-full"
          >
            Continuer
          </Button>
        </div>
      </div>
    </div>
  ),
  parameters: {
    layout: "fullscreen",
  },
}

// États de chargement
export const LoadingStates: Story = {
  render: () => (
    <div className="mx-auto max-w-4xl p-8">
      <h2 className="mb-6 text-2xl font-bold text-foreground">
        États de Chargement
      </h2>

      <div className="grid grid-cols-1 gap-8 md:grid-cols-3">
        <div className="space-y-4">
          <h4 className="font-semibold text-foreground">Chargement Simple</h4>
          <div className="space-y-3">
            <Button loading color="primary" size="lg" className="w-full">
              Chargement...
            </Button>
            <Button loading color="secondary" className="w-full">
              En cours
            </Button>
            <Button loading color="success" size="sm" className="w-full">
              Traitement
            </Button>
          </div>
        </div>

        <div className="space-y-4">
          <h4 className="font-semibold text-foreground">
            Avec Texte Personnalisé
          </h4>
          <div className="space-y-3">
            <Button
              loading
              loadingText="Sauvegarde..."
              color="primary"
              variant="shadow"
              className="w-full"
            >
              Sauvegarder
            </Button>
            <Button
              loading
              loadingText="Envoi..."
              color="success"
              variant="bordered"
              className="w-full"
            >
              Envoyer
            </Button>
            <Button
              loading
              loadingText="Connexion..."
              color="warning"
              variant="flat"
              className="w-full"
            >
              Se connecter
            </Button>
          </div>
        </div>

        <div className="space-y-4">
          <h4 className="font-semibold text-foreground">Avec Icônes</h4>
          <div className="space-y-3">
            <Button
              loading
              loadingText="Téléchargement..."
              leftIcon={<IconDownload size={16} />}
              color="primary"
              className="w-full"
            >
              Télécharger
            </Button>
            <Button
              loading
              loadingText="Partage..."
              leftIcon={<IconShare size={16} />}
              color="secondary"
              variant="bordered"
              className="w-full"
            >
              Partager
            </Button>
            <Button
              loading
              loadingText="Validation..."
              color="success"
              variant="light"
              className="w-full"
            >
              Valider
            </Button>
          </div>
        </div>
      </div>
    </div>
  ),
  parameters: {
    layout: "fullscreen",
  },
}
