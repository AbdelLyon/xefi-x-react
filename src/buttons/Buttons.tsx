import type { JSX } from "react";
import type { ButtonGroupProps } from "@heroui/react";
import { ButtonGroup } from "@heroui/react";
import type { ButtonProps } from "@/button/Button";
import { Button } from "@/button/Button";

export interface ButtonsProps extends ButtonGroupProps {
  buttons: Array<{
    key: string | number;
    label: React.ReactNode;
    buttonProps?: ButtonProps;
  }>;
}

export const Buttons = ({ buttons, ...props }: ButtonsProps): JSX.Element => {
  return (
    <ButtonGroup {...props}>
      {buttons.map(
        ({ key, label, buttonProps }): JSX.Element => (
          <Button key={key} {...buttonProps}>
            {label}
          </Button>
        ),
      )}
    </ButtonGroup>
  );
};
