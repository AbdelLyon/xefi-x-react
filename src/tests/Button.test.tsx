import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Button } from "@/button";
import { Buttons } from "@/buttons";
import userEvent from "@testing-library/user-event";
import type { JSX } from "react";
import { forwardRef } from "react";

export const MockLink = forwardRef<
  HTMLAnchorElement,
  React.AnchorHTMLAttributes<HTMLAnchorElement>
>(
  ({ children, ...props }, ref): JSX.Element => (
    <a ref={ref} {...props}>
      {children}
    </a>
  ),
);

MockLink.displayName = "MockLink";

describe("Composant Button", (): void => {
  describe("Snapshots", (): void => {
    it("devrait correspondre au snapshot avec contenu de base", (): void => {
      const { container } = render(<Button>Test</Button>);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec props fullWidth", (): void => {
      const { container } = render(<Button fullWidth>Test</Button>);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot en état de chargement", (): void => {
      const { container } = render(<Button isLoading>Test</Button>);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec contenu additionnel", (): void => {
      const { container } = render(
        <Button startContent={<span>Start</span>} endContent={<span>End</span>}>
          Test
        </Button>,
      );
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot en tant que lien", (): void => {
      const { container } = render(
        <Button href="/test" LinkComponent={MockLink}>
          Test Link
        </Button>,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Comportements", (): void => {
    it("devrait appeler onClick quand cliqué", async (): Promise<void> => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      const { getByRole } = render(<Button onClick={handleClick}>Test</Button>);
      const button = getByRole("button");

      await user.click(button);
      expect(handleClick).toHaveBeenCalledTimes(1);
    });

    it("ne devrait pas déclencher onClick si isLoading est true", async (): Promise<void> => {
      const handleClick = vi.fn();
      const user = userEvent.setup();

      const { getByRole } = render(
        <Button onClick={handleClick} isLoading>
          Test
        </Button>,
      );
      const button = getByRole("button");

      await user.click(button);
      expect(handleClick).not.toHaveBeenCalled();
    });
  });
});

describe("Composant Buttons", (): void => {
  const mockButtons = [
    { key: "1", label: "Bouton 1", buttonProps: { onClick: vi.fn() } },
    { key: "2", label: "Bouton 2", buttonProps: { isDisabled: true } },
  ];

  describe("Snapshots", (): void => {
    it("devrait correspondre au snapshot avec plusieurs boutons", (): void => {
      const { container } = render(<Buttons buttons={mockButtons} />);
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec groupe de boutons personnalisé", (): void => {
      const { container } = render(
        <Buttons
          buttons={mockButtons}
          className="w-full"
          aria-label="Groupe de boutons"
        />,
      );
      expect(container).toMatchSnapshot();
    });

    it("devrait correspondre au snapshot avec contenu React complexe", (): void => {
      const buttonsWithReactContent = [
        {
          key: "1",
          label: <span data-testid="custom-content">Label complexe</span>,
        },
      ];

      const { container } = render(
        <Buttons buttons={buttonsWithReactContent} />,
      );
      expect(container).toMatchSnapshot();
    });
  });

  describe("Comportements", (): void => {
    it("devrait appeler onClick du bouton", async (): Promise<void> => {
      const user = userEvent.setup();

      render(<Buttons buttons={mockButtons} />);
      const button = screen.getByText("Bouton 1");

      await user.click(button);
      expect(mockButtons[0].buttonProps?.onClick).toHaveBeenCalledTimes(1);
    });

    it("devrait gérer un groupe de boutons vide", (): void => {
      const { container } = render(<Buttons buttons={[]} />);
      expect(container.querySelectorAll("button")).toHaveLength(0);
    });
  });

  describe("Accessibilité", (): void => {
    it("devrait inclure les attributs ARIA", (): void => {
      render(<Buttons buttons={mockButtons} aria-label="Groupe de boutons" />);
      const group = screen.getByRole("group");
      expect(group).toHaveAttribute("aria-label", "Groupe de boutons");
    });
  });
});
