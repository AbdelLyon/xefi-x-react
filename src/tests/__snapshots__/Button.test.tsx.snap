// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Composant Button > Snapshots > devrait correspondre au snapshot avec contenu additionnel 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
    tabindex="0"
    type="button"
  >
    <span
      aria-hidden="true"
      focusable="false"
    >
      Start
    </span>
    Test
    <span
      aria-hidden="true"
      focusable="false"
    >
      End
    </span>
  </button>
</div>
`;

exports[`Composant Button > Snapshots > devrait correspondre au snapshot avec contenu de base 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
    tabindex="0"
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Composant Button > Snapshots > devrait correspondre au snapshot avec props fullWidth 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover w-full"
    tabindex="0"
    type="button"
  >
    Test
  </button>
</div>
`;

exports[`Composant Button > Snapshots > devrait correspondre au snapshot en tant que lien 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
    href="/test"
    tabindex="0"
    type="button"
  >
    Test Link
  </button>
</div>
`;

exports[`Composant Button > Snapshots > devrait correspondre au snapshot en état de chargement 1`] = `
<div>
  <button
    class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium opacity-disabled pointer-events-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground data-[hover=true]:opacity-hover"
    data-disabled="true"
    data-loading="true"
    disabled=""
    type="button"
  >
    <div
      aria-label="Loading"
      class="relative inline-flex flex-col gap-2 items-center justify-center"
    >
      <div
        class="relative flex w-5 h-5"
      >
        <i
          class="absolute w-full h-full rounded-full border-2 border-b-current animate-spinner-ease-spin border-solid border-t-transparent border-l-transparent border-r-transparent"
        />
        <i
          class="absolute w-full h-full rounded-full border-2 border-b-current opacity-75 animate-spinner-linear-spin border-dotted border-t-transparent border-l-transparent border-r-transparent"
        />
      </div>
    </div>
    Test
  </button>
</div>
`;

exports[`Composant Buttons > Snapshots > devrait correspondre au snapshot avec contenu React complexe 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center h-auto"
    role="group"
  >
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 [&:not(:first-child):not(:last-child)]:rounded-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground rounded-none first:rounded-s-medium last:rounded-e-medium data-[hover=true]:opacity-hover"
      tabindex="0"
      type="button"
    >
      <span
        data-testid="custom-content"
      >
        Label complexe
      </span>
    </button>
  </div>
</div>
`;

exports[`Composant Buttons > Snapshots > devrait correspondre au snapshot avec groupe de boutons personnalisé 1`] = `
<div>
  <div
    aria-label="Groupe de boutons"
    class="inline-flex items-center justify-center h-auto w-full"
    role="group"
  >
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 [&:not(:first-child):not(:last-child)]:rounded-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground rounded-none first:rounded-s-medium last:rounded-e-medium data-[hover=true]:opacity-hover"
      tabindex="0"
      type="button"
    >
      Bouton 1
    </button>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 opacity-disabled pointer-events-none [&:not(:first-child):not(:last-child)]:rounded-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground rounded-none first:rounded-s-medium last:rounded-e-medium data-[hover=true]:opacity-hover"
      data-disabled="true"
      disabled=""
      type="button"
    >
      Bouton 2
    </button>
  </div>
</div>
`;

exports[`Composant Buttons > Snapshots > devrait correspondre au snapshot avec plusieurs boutons 1`] = `
<div>
  <div
    class="inline-flex items-center justify-center h-auto"
    role="group"
  >
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 [&:not(:first-child):not(:last-child)]:rounded-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground rounded-none first:rounded-s-medium last:rounded-e-medium data-[hover=true]:opacity-hover"
      tabindex="0"
      type="button"
    >
      Bouton 1
    </button>
    <button
      class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent transform-gpu data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 opacity-disabled pointer-events-none [&:not(:first-child):not(:last-child)]:rounded-none [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary text-primary-foreground rounded-none first:rounded-s-medium last:rounded-e-medium data-[hover=true]:opacity-hover"
      data-disabled="true"
      disabled=""
      type="button"
    >
      Bouton 2
    </button>
  </div>
</div>
`;
